### PR TITLE
Docs: added self-managed connectors docker image documentation

### PR DESCRIPTION
**Pull request by bochenekmartin**
Issue: https://github.com/redpanda-data/documentation/issues/1611

### CODE REVIEW

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 1]
```diff
@@ -0,0 +1,192 @@
+---
+title: Connectors docker image
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:2]:
```suggestion
title: Connectors Docker Image
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+:::tip
+The general recommendation is to create a separate child directory for each connector, and to place that connector's JAR files and other resource files in that child directory.
+:::
+
+## Configure SASL
+
+1. Prepare SASL user and password, make sure the user has right permissions.
+* Write access for internal topics and access to consumer group (so all workers in the cluster can communicate each other) is always required.
+* ACLs depend on used connector type (source/sink) and topics used by the connectors.
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:70]:
```suggestion
1. Prepare the SASL user and password, making sure the user has necessary permissions.
- Write access for internal topics and access to consumer groups (so all workers in the cluster can communicate with each other) is required.
- ACLs depend on used connector type (source/sink) and topics used by the connectors.
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+2. Create a file containing the plain-text password in a dedicated directory. For example in `./connect-password/redpanda-password/password` where `password` file contains just the password.
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:71]:
```suggestion
2. Create a file containing the plain text password in a dedicated directory. For example, in `./connect-password/redpanda-password/password` where the `password` file contains just the password
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+3. Mount a volume to bind the directory with a container. For example make `./connect-password` directory content visible in `/opt/kafka/connect-password` in a container:
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:72]:
```suggestion
3. Mount a volume to bind the directory with a container. For example, make the `./connect-password` directory content visible in `/opt/kafka/connect-password` in a container:
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+```yaml
+volumes:
+  - ./connect-password:/opt/kafka/connect-password
+```
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:76]:
```suggestion
  ```yaml
  volumes:
    - ./connect-password:/opt/kafka/connect-password
  ```
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+title: Connectors Docker Image
+---
+
+<head>
+    <meta name="title" content="Docker image | Redpanda Docs"/>
+    <meta name="description" content="Connectors self-hosted docker image"/>
+    <link rel="canonical" href="https://docs.redpanda.com/docs/deploy/deployment-option/self-hosted/connectors/docker-image/" />
+</head>
+
+The image is available in a Docker Hub repository: https://hub.docker.com/r/redpandadata/connectors/tags
```
Comment (Feediver1) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:11]:
```suggestion

```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+* JMX-Exporter JMX Prometheus JavaAgent
+
+Included connectors:
+* MirrorSourceConnector
+* MirrorCheckpointConnector
+* MirrorHeartbeatConnector
+
+
+## Image configuration properties
+| Property                           | Meaning                                                                                                                                                                                                                                                                                                                                               |
```
Comment (Feediver1) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:26]:
```suggestion
| Property                           | Description                                                                                                                                                                                                                                                                                                                                               |
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+* Kafka Connect
+* JMX-Exporter JMX Prometheus JavaAgent
+
+Included connectors:
+* MirrorSourceConnector
+* MirrorCheckpointConnector
+* MirrorHeartbeatConnector
+
+
+## Image configuration properties
```
Comment (Feediver1) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:25]:
```suggestion
## Docker image configuration properties

The following table lists the available Docker image properties.
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+volumes:
+  - ./connect-certs:/opt/kafka/connect-certs/user-secret
+```
+3. Set `CONNECT_TLS_ENABLED` property to `"true"`.
+4. Use `CONNECT_TLS_AUTH_CERT`, `CONNECT_TRUSTED_CERTS` and `CONNECT_TLS_AUTH_KEY` image properties to configure relative path to cert and key. For example, if files are in `/opt/kafka/connect-certs/user-secret`, use:
+```yaml
+CONNECT_TRUSTED_CERTS: "user-secret/ca.crt"
+CONNECT_TLS_AUTH_CERT: "user-secret/client.crt"
+CONNECT_TLS_AUTH_KEY: "user-secret/client.key"
+```
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:102]:
```suggestion
4. Use the `CONNECT_TLS_AUTH_CERT`, `CONNECT_TRUSTED_CERTS`, and `CONNECT_TLS_AUTH_KEY` image properties to configure the relative path to the certificate and key. For example, if the files are in `/opt/kafka/connect-certs/user-secret`, use:
  ```yaml
  CONNECT_TRUSTED_CERTS: "user-secret/ca.crt"
  CONNECT_TLS_AUTH_CERT: "user-secret/client.crt"
  CONNECT_TLS_AUTH_KEY: "user-secret/client.key"
  ```
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+    │   └── redpanda-password
+    │       └──password                            # A file with SASL password
+    ├── connect-plugins
+    │   └── custom-connector
+    │       └── custom-sink-connector-fat.jar      # Connector fat jar or jar and dependencies jars
+    └── docker-compose.yaml                        # A docker-compose file
+
+1. Execute `docker-compose up` command from a directory containing `docker-compose.yaml` file.
+2. List installed plugins: `curl localhost:8083/connector-plugins`.
+3. Get Kafka Connect basic info: `curl localhost:8083/`.
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:157]:
```suggestion
3. To get Kafka Connect basic information, run: `curl localhost:8083/`.
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+4. Metrics are available at `localhost:9404/`.
+5. Use Redpanda Console or Kafka Connect REST API to manage connectors.
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:159]:
```suggestion
5. Use the Redpanda Console or Kafka Connect REST API to manage connectors.
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+
+## REST API
+It is possible to manage connectors using Kafka Connect REST API.
+Examples:
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:163]:
```suggestion
You can manage connectors using the Kafka Connect REST API. For example:
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+*  Get version of Kafka Connect worker
+```text
+curl localhost:8083 | jq
+```
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:167]:
```suggestion
To get the version of the Kafka Connect worker:
```text
curl localhost:8083 | jq
```
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 14]
```diff
@@ -0,0 +1,192 @@
...
+CONNECT_PLUGIN_PATH: "/opt/kafka/connect-plugins"
+```
+4. New connector type should be discovered by Kafka Connect automatically on startup. Use `/connector-plugins` Kafka Connect REST endpoint to check available connector types list. For example: `curl localhost:8083/connector-plugins`
+
+:::tip
+The general recommendation is to create a separate child directory for each connector, and to place that connector's JAR files and other resource files in that child directory.
+:::
+
+## Configure SASL
+
```
Comment (Feediver1) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:67]: To configure SASL:

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 15]
```diff
@@ -0,0 +1,192 @@
...
+## REST API
+It is possible to manage connectors using Kafka Connect REST API.
+Examples:
+*  Get version of Kafka Connect worker
+```text
+curl localhost:8083 | jq
+```
+
+* Get list of available connector plugins
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:169]:
```suggestion
To get the list of available connector plugins:
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 16]
```diff
@@ -0,0 +1,192 @@
...
+```text
+curl localhost:8083/connector-plugins | jq
+```
+* Get list of active connectors
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:173]:
```suggestion
To get the list of active connectors:
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 17]
```diff
@@ -0,0 +1,192 @@
...
+```text
+curl 'http://localhost:8083/connectors?expand=status&expand=info' | jq
+```
+* Create connector `curl "localhost:8083/connectors" -H 'Content-Type: application/json' --data-raw '<connector-config>'`
+for example
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:178]:
```suggestion
To create the connector `curl "localhost:8083/connectors" -H 'Content-Type: application/json' --data-raw '<connector-config>'`:
```

File: docs/deploy/deployment-option/self-hosted/connectors/index.mdx [hunk 18]
```diff
@@ -0,0 +1,13 @@
...
+
+<head>
+    <meta name="title" content="Deploy Connectors | Redpanda Docs"/>
+    <meta name="description" content="Deploy Connectors."/>
+    <link rel="canonical" href="https://docs.redpanda.com/docs/deploy/deployment-option/self-hosted/connectors/" />
+</head>
+
+- [Docker image](../connectors/docker-image/)
+  
+    Follow the guidelines to deploy Connectors Docker image.
```
Comment (Feediver1) [inline docs/deploy/deployment-option/self-hosted/connectors/index.mdx:13]:
```suggestion
   Guidelines for deploying the Connectors Docker image. 
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 19]
```diff
@@ -0,0 +1,192 @@
...
+  -H 'Content-Type: application/json' \
+  --data-raw '{  "name": "heartbeat-connector", "config": { "connector.class": "org.apache.kafka.connect.mirror.MirrorHeartbeatConnector", "heartbeats.topic.replication.factor": "1", "replication.factor": "1",  "source.cluster.alias": "source",  "source.cluster.bootstrap.servers": "redpanda:29092",     "target.cluster.bootstrap.servers": "redpanda:29092"}}'
+```
+* Get connector status `curl localhost:8083/connectors/<connector-name>/status`
+for example:
+```text
+curl localhost:8083/connectors/heartbeat-connector/status
+```
+* Delete connector `curl "localhost:8083/connectors/<connector-name>" -X 'DELETE'` for example:
+```text
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:190]:
```suggestion
To delete the connector `curl "localhost:8083/connectors/<connector-name>" -X 'DELETE'`:
```text
```

File: docs/deploy/deployment-option/self-hosted/connectors/index.mdx (hunk 18 above)
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/index.mdx:13]:
```suggestion
    Follow the guidelines to deploy the connectors Docker image. 
```

File: docs/deploy/deployment-option/self-hosted/index.mdx [hunk 20]
```diff
@@ -15,4 +15,8 @@ title: Self-Hosted Deployment
 
 - [Linux](../self-hosted/manual)
 
-   Learn about deployment options on Linux, as well as considerations for high availability and sizing. 
\ No newline at end of file
+   Learn about deployment options on Linux, as well as considerations for high availability and sizing. 
+
+- [Connectors](../self-hosted/connectors)
+
+   Learn about Connectors deployment options.
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/index.mdx:22]:
```suggestion
   Learn about connectors deployment options. 
```

**Review by micheleRP — COMMENTED** [review]
@Feediver1 please see initial review

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 21]
```diff
@@ -0,0 +1,192 @@
+---
+title: Connectors Docker Image
```
Comment (Feediver1) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:2]:
```suggestion
title: Configure Connectors with Docker Image
```

File: sidebars.js [hunk 22]
```diff
@@ -283,6 +283,21 @@
                     },
                   ]
                 },
+                {
+                  "type": "category",
+                  "label": "Connectors",
```
Comment (Feediver1) [inline sidebars.js:288]:
```suggestion
                  "label": "Configure Connectors with Docker Image",
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 23]
```diff
@@ -0,0 +1,192 @@
...
+```
+3. Set `CONNECT_TLS_ENABLED` property to `"true"`.
+4. Use `CONNECT_TLS_AUTH_CERT`, `CONNECT_TRUSTED_CERTS` and `CONNECT_TLS_AUTH_KEY` image properties to configure relative path to cert and key. For example, if files are in `/opt/kafka/connect-certs/user-secret`, use:
+```yaml
+CONNECT_TRUSTED_CERTS: "user-secret/ca.crt"
+CONNECT_TLS_AUTH_CERT: "user-secret/client.crt"
+CONNECT_TLS_AUTH_KEY: "user-secret/client.key"
+```
+
+#### Sample Docker-compose file
```
Comment (Feediver1) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:104]:
```suggestion
#### Sample Docker Compose file
```

File: docs/deploy/deployment-option/self-hosted/index.mdx (hunk 20 above)
Comment (Feediver1) [reply docs/deploy/deployment-option/self-hosted/index.mdx:22]: Learn about connector deployment options.

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 24]
```diff
@@ -0,0 +1,193 @@
...
+curl "localhost:8083/connectors" \
+  -H 'Content-Type: application/json' \
+  --data-raw '{  "name": "heartbeat-connector", "config": { "connector.class": "org.apache.kafka.connect.mirror.MirrorHeartbeatConnector", "heartbeats.topic.replication.factor": "1", "replication.factor": "1",  "source.cluster.alias": "source",  "source.cluster.bootstrap.servers": "redpanda:29092",     "target.cluster.bootstrap.servers": "redpanda:29092"}}'
+```
+* Get connector status `curl localhost:8083/connectors/<connector-name>/status`
+for example:
+```text
+curl localhost:8083/connectors/heartbeat-connector/status
+```
+To delete the connector `curl "localhost:8083/connectors/<connector-name>" -X 'DELETE'`:
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:190]:
```suggestion
To delete the connector, run:  
```
curl "localhost:8083/connectors/<connector-name>" -X 'DELETE'
```
For example:
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 25]
```diff
@@ -0,0 +1,192 @@
...
+```text
+curl "localhost:8083/connectors" \
+  -H 'Content-Type: application/json' \
+  --data-raw '{  "name": "heartbeat-connector", "config": { "connector.class": "org.apache.kafka.connect.mirror.MirrorHeartbeatConnector", "heartbeats.topic.replication.factor": "1", "replication.factor": "1",  "source.cluster.alias": "source",  "source.cluster.bootstrap.servers": "redpanda:29092",     "target.cluster.bootstrap.servers": "redpanda:29092"}}'
+```
+* Get connector status `curl localhost:8083/connectors/<connector-name>/status`
+for example:
+```text
+curl localhost:8083/connectors/heartbeat-connector/status
+```
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:188]:
```suggestion
To get connector status, run: 
```text
curl localhost:8083/connectors/<connector-name>/status
```
For example:
```text
curl localhost:8083/connectors/heartbeat-connector/status
```
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 26]
```diff
@@ -0,0 +1,181 @@
...
+  volumes:
+    - ./connect-password:/opt/kafka/connect-password
+4. Use `CONNECT_SASL_USERNAME` to set the SASL username, and use `CONNECT_SASL_PASSWORD_FILE` to set the relative path to a password file. For example, if the file is in `/opt/kafka/connect-password/redpanda-password/password`, use the `redpanda-password/password` value.
+  ```yaml
+  CONNECT_SASL_USERNAME: "connect-user"
+  CONNECT_SASL_PASSWORD_FILE: "redpanda-password/password"
+
+## Configure TLS
+
+1. Prepare Redpanda cluster cert and key. Place them in a dedicated directory, for example:
```
Comment (Deflaimun) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:83]: 1. Prepare the Redpanda cluster certificate and key, and place them in a dedicated directory. For example:
```
    ./connect-certs/ca.crt
    ./connect-certs/client.crt
    ./connect-certs/client.key
```

Comment (Deflaimun) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:83]:
```suggestion
1. Prepare the Redpanda cluster certificate and key, and place them in a dedicated directory. For example:
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 27]
```diff
@@ -0,0 +1,181 @@
...
+```
+./connect-certs/ca.crt
+./connect-certs/client.crt
+./connect-certs/client.key
```
Comment (Deflaimun) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:87]:
```suggestion
      ./connect-certs/ca.crt
      ./connect-certs/client.crt
      ./connect-certs/client.key
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 28]
```diff
@@ -0,0 +1,181 @@
...
+  volumes:
+    - ./connect-plugins:/opt/kafka/connect-plugins
+3. Use the `CONNECT_PLUGIN_PATH` image property to configure a directory with the new connector. Use Kafka Connect to discover new connectors. For example:
+  ```yaml
+  CONNECT_PLUGIN_PATH: "/opt/kafka/connect-plugins"
+4. The new connector type should be discovered by Kafka Connect automatically on startup. Use the `/connector-plugins` Kafka Connect REST endpoint to check available connector types. For example: `curl localhost:8083/connector-plugins`
+
+:::tip
+Create a separate child directory for each connector, and place the connector's jar files and other resource files in that child directory.
+:::
```
Comment (micheleRP) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:64]:
```suggestion
  :::tip
  Create a separate child directory for each connector, and place the connector's jar files and other resource files in that child directory.
  :::
```

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 29]
```diff
@@ -0,0 +1,259 @@
...
+
+  ```yaml
+  CONNECT_TRUSTED_CERTS: "user-secret/ca.crt"
+  CONNECT_TLS_AUTH_CERT: "user-secret/client.crt"
+  CONNECT_TLS_AUTH_KEY: "user-secret/client.key"
+  ```
+
+## Connect with Docker Compose
+
+The following Docker Compose file can be used as a sample:
```
Comment (Feediver1) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:124]:
```suggestion
You can use the following Docker Compose file as a sample:
```

File: sidebars.js [hunk 30]
```diff
@@ -283,6 +283,21 @@
...
+                  "link": {
+                    "type": "doc",
+                    "id": "deploy/deployment-option/self-hosted/connectors/index"
```
Comment (Feediver1) [inline sidebars.js:291]: I think we agreed to just have a single entry: Configure Connectors with Docker Compose

File: docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx [hunk 31]
```diff
@@ -0,0 +1,259 @@
...
+    │   └── client.key                             # A file with Redpanda cluster key
+    ├── connect-password
+    │   └── redpanda-password
+    │       └──password                            # A file with SASL password
+    ├── connect-plugins
+    │   └── custom-connector
+    │       └── custom-sink-connector-fat.jar      # Connector fat jar or jar and dependencies jars
+    └── docker-compose.yaml                        # A docker-compose file
+
+1. Run the `docker-compose up` command from a directory containing the `docker-compose.yaml` file.
```
Comment (Feediver1) [inline docs/deploy/deployment-option/self-hosted/connectors/docker-image.mdx:176]: These steps need an introductory statement providing context. Or, move under a heading?

**Review by Feediver1 — APPROVED** [review]
Please address two issues:
- sidenav should be a single entry: "Configure Connectors with Docker Image"
- Steps under the sample in "Connect with Docker Compose" need an intro/context.
Once these are addressed. you can merge.

Awesome job helping out @micheleRP 
And thanks for the PR @bochenekmartin

**Pull request merged**